### PR TITLE
fix: Append the error message prior to offloading node status

### PR DESCRIPTION
**Pull request by terrytangyuan**
Currently if offloading is not enabled and if the workflow is too large, we will only see `offload node status is not supported` and the original error message is lost/masked. We'd like to expose that information so users will know their workflow is too large by how much and know what to do next.

Signed-off-by: terrytangyuan <terrytangyuan@gmail.com>

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

**Review by simster7 — DISMISSED** [review]

File: workflow/hydrator/hydrator.go [hunk 1]
```diff
@@ -101,12 +102,17 @@ func (h hydrator) Dehydrate(wf *wfv1.Workflow) error {
 	}
 	if packer.IsTooLargeError(err) || alwaysOffloadNodeStatus {
 		var offloadVersion string
-		err := waitutil.Backoff(writeRetry, func() (bool, error) {
```
Comment (alexec) [inline workflow/hydrator/hydrator.go:104]: it is impossible for this to happen AFAIK - alwaysOffloadNodeStatus must always false , so the error must be IsTooLargeError

**Review by alexec — CHANGES_REQUESTED** [review]
I don't understand this fix. This code branch should only even execute if the error is `workflow is longer than maximum allowed size.`.

**Comment by terrytangyuan** [comment]
@alexec We are observing error messages like `offload node status is not supported` in our workflows but the original error message is masked. Here's what the new error message looks like (it's the new test case I added in this PR):

`workflow is longer than maximum allowed size. compressed size 317 > maxSize 260.Tried to offload but encountered error: offload node status is not supported`

> I don't understand this fix. This code branch should only even execute if the error is `workflow is longer than maximum allowed size.`.

Perhaps I misunderstood something here but the code would also execute when `alwaysOffloadNodeStatus == true`.

**Review by alexec — CHANGES_REQUESTED** [review]
Please sync with master.

**Comment by terrytangyuan** [comment]
@alexec Rebased but outdated again. I thought rebasing is done after approved and before merging?

**Review by alexec — APPROVED** [review]

**Pull request merged**